libavfilter/qsvvpp.c

```diff
@@ -300,10 +300,12 @@ static QSVFrame *submit_frame(QSVVPPContext *s, AVFilterLink *inlink, AVFrame *p
         qsv_frame->surface = (mfxFrameSurface1 *)qsv_frame->frame->data[3];
     } else {
         /* make a copy if the input is not padded as libmfx requires */
-        if (picref->height & 31 || picref->linesize[0] & 31) {
+        const AVPixFmtDescriptor *desc = av_pix_fmt_desc_get(picref->format);
+        if (picref->height & 31 || picref->linesize[0] & 31
+            || (!strcmp(desc->name, "yuv420p") && picref->linesize[0] & 63)) {
```

Comment:
Only yuv420p format needs this patch or all format of system memory?  And currently your patch will change the original logic for the case "not yuv420p , 32bytes aligned but not 64 bytes aligned".

Reply:
Hi, 
FFmpeg aligns linesize of all components (Y,U,V) with 32 bits separately. When width = 720, after alignment the Y’s linesize is 736, and U, V’s line size is 384 and “736 != 384 * 2”. But mfxFrameSurface1 in current mSDK has only one linesize field for Y component; and I think it use half of Y’s linesize to be U, V’s linesize, and it’s the root cause of the issue.

     When  using both HW Decoder and encoder, the frame buffer is vasurface and use NV12 format, and FFmpeg’s alignment won’t be used. So there is no such issue.

Reply:
It makes sense.  but I think it should be change to FFALIGN(inlink->w, 64), and no change for inlink->h.

Reply:
if it works no problem

Reply:
Hi, any feedback ?